R/auth_sitrep.R

```diff
@@ -0,0 +1,187 @@
+#' Twitter Tokens sitrep
+#' 
+#' Get a situation report of your current tokens; useful for upgrading from 
+#' rtweet 0.7.0 to 1.0.0 and diagnosing problems with tokens. Called by it side effects
+#' 
+#' Searches for old tokens on the user folder, if duplicate tokens are found 
+#' they are deleted. Tokens are then moved to the new location. 
+#' @return Invisible the path of the tokens
+#' @export
+#' @examples
+#' auth_sitrep()
+auth_sitrep <- function() {
+  all_tokens_files <- c(find_old_tokens(), 
+                        find_rappdirs_tokens(), 
+                        find_tools_tokens())
+  path <- tools::R_user_dir("rtweet", "config")
+  if (!dir.exists(path)) {
+    dir.create(path, showWarnings = FALSE, recursive = TRUE)
+  }
+  
+  if (is.null(all_tokens_files)) {
+    inform("No tokens where found! See auth_as help file.")
+    return(NULL)
+  }
+  
+  type_auth <- type_auth(all_tokens_files)
+  all_tokens <- lapply(all_tokens_files, readRDS)
+  names(all_tokens) <- all_tokens_files
+  
+  if (any(type_auth == "bearer")){
+    handle_bearer(all_tokens[type_auth == "bearer"], path)
+  }
+  if (any(type_auth == "token")){
+    handle_token(all_tokens[type_auth == "token"], path)
+  }
+  invisible(list.files(path, pattern = "*.rds", all.files = TRUE, 
+                       full.names = TRUE))
+}
+
+find_old_tokens <- function() {
+  twitter_pat <- Sys.getenv("TWITTER_PAT")
+  home_path <- normalizePath(file.path("~"), mustWork = TRUE)
+  
+  many_paths <- c(twitter_pat, home_path)
+  old_tokens <- lapply(many_paths, list.files, pattern = ".rtweet_token.*rds", 
+                       full.names = TRUE, all.files = TRUE)
+  unlist(old_tokens, TRUE, FALSE)
+}
+
+find_rappdirs_tokens <- function() {
+  if (requireNamespace("rappdirs", quietly = TRUE)) {
+    rappdirs_path <- getOption("rtweet:::config_dir", 
+                               rappdirs::user_config_dir("rtweet"))
+    rappdirs_tokens <- list.files(rappdirs_path, pattern = "*.rds", 
+                                  all.files = TRUE, full.names = TRUE)
+  } else {
+    rappdirs_tokens <- NULL
+  }
+  rappdirs_tokens
+}
+
+find_tools_tokens <- function() {
+  path <- tools::R_user_dir("rtweet", "config")
+  final_path <- getOption("rtweet:::config_dir", path)
+  if (dir.exists(final_path)) {
```

Comment:
You don't actually need this check here since list files will return an empty character vector if the path doesn't exist.